### ISSUE TITLE
How to add CNAME for custom siteURL

### ISSUE THREAD

**Issue by dfvr1994** [Azure/Moodle#225]
Hi, two questions,

I've setup a Moodle resource group using the custom template and only changed the siteURL parameter pointing to a DNS I just purchased. 

1) How do I add the CNAME record on goddaddy? 

![image](https://user-images.githubusercontent.com/45764991/101551240-bdecb000-397e-11eb-93b6-b5fd9ef0065a.png)

2) I did some testing before by deploying Moodle without adding a custom siteURL. I could point to either the load balancer Public IP or the load balancer DNS and land on Moodle's homepage. 

Now if I navigate to the load balancer IP or DNS I get redirected to my custom siteURL and get: ERR_CONNECTION_CLOSED

Is it because of the CNAME missing? Could someone please explain how the load balancer public IPs and CNAME record setup on godaddy interact with each other? If everything is setup corrrectly will I see my custom siteURL DNS on top of my browser, or will I see the load balancer DNS?

Thanks

**Comment by Zhuich** [comment]
You need to use SiteURL parameter in moodle deploying template

**Comment by dfvr1994** [comment]
@Zhuich I as I mentioned before I did use the SiteURL parameter in the template, with no success. Problem is that if I navigate to the load balancer IP or DNS I get redirected to my custom siteURL and get: ERR_CONNECTION_CLOSED

Could you please elaborate your answer? How does the load balancer public IPs and CNAME record setup on godaddy interact with each other? If everything is setup correctly will I see my custom siteURL DNS on top of my browser, or will I see the load balancer DNS?

Thanks

**Comment by SorraTheOrc** [comment]
@dfvr1994 While this template will do all the necessary configuration on the Azure side we cannot setup the DNS for you as it is managed by whatever Domain Name Registrar you have chosen. 

You can setup your domain to be managed on Azure, see https://docs.microsoft.com/en-us/azure/dns/dns-delegate-domain-azure-dns or you need to follow the instructions for configuring a CNAME with your domain name provider. You mention Godaddy so you should start at https://www.godaddy.com/help/add-a-cname-record-19236

In order to complete this on Godaddy you need to point your CName to your LoadBalancerDNS (or IP). You can get these values from the portal or from the CLI as described in https://github.com/Azure/Moodle/blob/b6dfa0c11b605f42e7c5e8a0c6e2df70461b05ef/docs/Get-Install-Data.md

**Comment by dfvr1994** [comment]
Hi @SorraTheOrc thanks for elaborating. Is this the way I should add the CNAME record? 

![image](https://user-images.githubusercontent.com/45764991/104787080-a0b8b780-575c-11eb-82eb-7468905be29a.png)

??

**Comment by SorraTheOrc** [comment]
It depends on what the correct settings are for your deployment. What is the output of the command `az group deployment show --resource-group $MOODLE_RG_NAME --name $MOODLE_DEPLOYMENT_NAME --out json --query *.outputs`?

If you are not using the CLI what were the inputs you provided when deploying the cluster?
________________________________

**Comment by dfvr1994** [comment]
@SorraTheOrc I used the full customizable template found here:

https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FMoodle%2Fmaster%2Fazuredeploy.json

I set HTTPS Termination to none, and provided my custom siteURL and SSH Public Key:

Result of the command is: (replaced passwords for obvious reasons)

```[
  {
    "controllerInstanceIP": {
      "type": "String",
      "value": "40.78.46.218"
    },
    "databaseAdminPassword": {
      "type": "String",
      "value": "MY_PW"
    },
    "databaseAdminUsername": {
      "type": "String",
      "value": "dbadmin@mysql-fwp74o"
    },
    "databaseDNS": {
      "type": "String",
      "value": "mysql-fwp74o.mysql.database.azure.com"
    },
    "firstFrontendVmIP": {
      "type": "String",
      "value": "172.31.0.5"
    },
    "loadBalancerDNS": {
      "type": "String",
      "value": "lb-fwp74o.westus.cloudapp.azure.com"
    },
    "loadBalancerSku": {
      "type": "String",
      "value": "Basic"
    },
    "moodleAdminPassword": {
      "type": "String",
      "value": "MY_PW"
    },
    "moodleDbPassword": {
      "type": "String",
      "value": "MY_PW"
    },
    "moodleDbUsername": {
      "type": "String",
      "value": "moodle@mysql-fwp74o"
    },
    "siteURL": {
      "type": "String",
      "value": "estudia-baciller.com"
    }
  }
]```

**Comment by SorraTheOrc** [comment]
Your Load balancer is: lb-fwp74o.westus.cloudapp.azure.com (this is the "points to" setting in the DNS registry)
Your site URL is: estudia-baciller.com (this is your HOST setting in the DNS registry)

**Comment by dfvr1994** [comment]
So I talked to GoDaddy support and ended up setting DNS records like this: 

![image](https://user-images.githubusercontent.com/45764991/105128287-ba396680-5ab0-11eb-8cc6-917a2e67854e.png)

@SorraTheOrc  They told me to add an A record pointing to the public IP of my website (the load balancer I guess). If I add "estudia-baciller.com" to the HOST I would end up with this CNAME record: "estudia-baciller.com.estudia-baciller.com". Therefore I set "www" to HOST and the load balancer DNS to POINTS TO value. Load balancer DNS is different now because I made a new deployment. 

Problem:

If I navigate to my website it seems like MOODLE images are not loading. I'm sure this is not a issue with my internet connection.  Could this be a result of not setting some DNS record correctly?

![image](https://user-images.githubusercontent.com/45764991/105128806-bf4ae580-5ab1-11eb-95db-6f49fd9bc992.png)
